### PR TITLE
Optimize StableLM

### PR DESCRIPTION
**Pull request by Oscilloscope98**
## Description

https://github.com/analytics-zoo/nano/issues/1233

- rope
- fused mlp
- fused layernorm
- merge qkv
- kv cache optimization

### CODE REVIEW

**Review by rnwang04 — APPROVED** [review]
LGTM

**Comment by Oscilloscope98** [comment]
Merge for now. Will Investigate on the logits difference caused by rope optimizations later.
